### PR TITLE
Fix WiFi and real installation.

### PR DESCRIPTION
**Pull request by MagneFire**
Turns out that the Moto 360 2015 has a very limited allowed boot image size...
The default size of the boot image was around 11.7MB which exceeded the allowed size (i.e. results in: `(bootloader) image size exceeded partition limits`).
I attempted to streamline the kernel some more. But with minimal results.
Removing android-tools and bash shaves around 2MB from the boot image, which is enough to make it flash to the boot partition.

Another issue solved is WiFi. For some reason. After unblocking WiFi the error `wl_host_event_get_data: Invalid event frame, err = -24` is spammed in the kernel log. Simple thing I tried was to use the Nougat firmware. This solves it :)

### CODE REVIEW

File: recipes-kernel/linux/linux-smelt/defconfig [hunk 1]
```diff
@@ -458,8 +455,7 @@ CONFIG_AEABI=y
 # CONFIG_ARCH_SPARSEMEM_DEFAULT is not set
 # CONFIG_ARCH_SELECT_MEMORY_MODEL is not set
 CONFIG_HAVE_ARCH_PFN_VALID=y
-CONFIG_HIGHMEM=y
-# CONFIG_HIGHPTE is not set
+# CONFIG_HIGHMEM is not set
```
Comment (FlorentRevest) [inline recipes-kernel/linux/linux-smelt/defconfig:458]: What's the motivation behind disabling CONFIG_HIGHMEM?

File: recipes-kernel/linux/linux-smelt/defconfig [hunk 2]
```diff
@@ -1165,8 +1159,8 @@ CONFIG_USB_HSIC_SMSC_HUB=y
 # CONFIG_QCOM_LIQUID_DOCK is not set
 CONFIG_MMI_FACTORY=y
 CONFIG_BQ5105X_CTRL=y
-CONFIG_MOT_UTAG=y
-CONFIG_SSV_GPS=y
+# CONFIG_MOT_UTAG is not set
+# CONFIG_SSV_GPS is not set
```
Comment (FlorentRevest) [inline recipes-kernel/linux/linux-smelt/defconfig:1163]: Is this not the integrated GPS in smelt?

File: recipes-kernel/linux/linux-smelt/defconfig [hunk 3]
```diff
@@ -3361,7 +3294,7 @@ CONFIG_ENCRYPTED_KEYS=y
 # CONFIG_SECURITY_DMESG_RESTRICT is not set
 # CONFIG_SECURITY is not set
 # CONFIG_SECURITYFS is not set
-# CONFIG_DEFAULT_SECURITY_DAC is not set
+CONFIG_DEFAULT_SECURITY_DAC=y
```
Comment (FlorentRevest) [inline recipes-kernel/linux/linux-smelt/defconfig:3297]: Why do you need this?

File: recipes-kernel/linux/linux-smelt/defconfig [hunk 4]
```diff
@@ -3531,31 +3464,3 @@ CONFIG_ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE=y
 CONFIG_QMI_ENCDEC=y
 # CONFIG_QMI_ENCDEC_DEBUG is not set
 # CONFIG_VIRTUALIZATION is not set
-CONFIG_IP_NF_MULTIPLE_TABLES=y
-CONFIG_CGROUP_MEM_RES_CTLR=y
-CONFIG_CGROUP_MEM_RES_CTLR_SWAP=y
-CONFIG_CGROUP_MEM_RES_CTLR_KMEM=y
-CONFIG_BLK_SCHED=y
```
Comment (FlorentRevest) [inline recipes-kernel/linux/linux-smelt/defconfig:3538]: I don't find the documentation for this but I imagine this is a downstream block scheduler, is there any reason for disabling it?

File: recipes-kernel/linux/linux-smelt/defconfig [hunk 5]
```diff
@@ -3531,31 +3464,3 @@ CONFIG_ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE=y
...
-CONFIG_DEBUG_RODATA=y
-CONFIG_DEVPTS_FS=y
```
Comment (FlorentRevest) [inline recipes-kernel/linux/linux-smelt/defconfig:3540]: Doesn't the ramdisk fail without this https://github.com/AsteroidOS/meta-asteroid/blob/master/recipes-core/initrdscripts/initramfs-scripts-android/init.sh#L24 ?

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 1 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:458]: Mostly assumed though, but here's the description what I found:
```
The address space of ARM processors is only 4 Gigabytes large and it has to accommodate user address space, kernel address space as well as some memory mapped IO. That means that, if you have a large amount of physical memory and/or IO, not all of the memory can be "permanently mapped" by the kernel. The physical memory that is not permanently mapped is called "high memory".

Depending on the selected kernel/user memory split, minimum vmalloc space and actual amount of RAM, you may not need this option which should result in a slightly faster kernel.
```

The Moto 360 only has 512 Megabytes of RAM, so I simply assumed that this option wouldn't be needed. Not entirely sure though.

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 2 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:1163]: Nope, only the sport edition has GPS.

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 3 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:3297]: Eh, not sure. Must have been automatically enabled itself. while messing with `menuconfig`.
My process on defconfig changes is:
```sh
bitbake -c menuconfig linux-smelt
# Adjust the kernel config
# Let it compile, test if it runs. look for obvious issues in dmesg.
# Copy from `work-shared` .config to defconfig
```

Will look into this.

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 4 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:3538]: See previous, this has probably happened when I disabled CGROUP debugging.

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 3 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:3297]: This is actually set by the Kconfig of security:
```
config DEFAULT_SECURITY
        string
        default "selinux" if DEFAULT_SECURITY_SELINUX
        default "smack" if DEFAULT_SECURITY_SMACK
        default "tomoyo" if DEFAULT_SECURITY_TOMOYO
        default "apparmor" if DEFAULT_SECURITY_APPARMOR
        default "yama" if DEFAULT_SECURITY_YAMA
        default "" if DEFAULT_SECURITY_DAC

endmenu
```

And we have `CONFIG_DEFAULT_SECURITY=""`

Comment (FlorentRevest) [reply recipes-kernel/linux/linux-smelt/defconfig:3297]: Mhh okay fair enough. We don't have CONFIG_SECURITY set anyway so there shouldn't be any LSM. That's fine

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 1 above)
Comment (FlorentRevest) [reply recipes-kernel/linux/linux-smelt/defconfig:458]: I imagine this is really a nano-optimization but if you didn't notice any weird behaviour I'm ok with it.

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 5 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:3540]: Hm, looks like this option doesn't exist...
I believe that the option in this kernel is named `CONFIG_UNIX98_PTYS` which is enabled, see: https://android.googlesource.com/kernel/msm/+/refs/heads/android-msm-smelt-3.10-marshmallow-mr1-wear-release/fs/devpts/Makefile

File: recipes-kernel/linux/linux-smelt/defconfig (hunk 4 above)
Comment (MagneFire) [reply recipes-kernel/linux/linux-smelt/defconfig:3538]: This is actually enabled by `check_config`, but Google doesn't even result in anything useful :-)

Comment (FlorentRevest) [reply recipes-kernel/linux/linux-smelt/defconfig:3538]: That will do

**Pull request merged**